internal/opts/config.go

```diff
@@ -115,6 +140,28 @@ func (c *Config) validateFlags() {
 			log.Panicf("given master address: %s for replication is invalid, must be in host:port format", c.ReplicationMasterAddr)
 		}
 	}
+
+	//validate discovery server config
+
+	discoveryServerConfig := c.DiscoveryServiceConfig
+
+	if c.DbRole == "discovery" {
+
+		if discoveryServerConfig.ServerConfig.HeartbeatTimeout <= 0 ||
+			discoveryServerConfig.ServerConfig.StatusTTl <= 0 {
+			log.Panicf("Invalid discovery server configuration")
+		}
+	}
+
+	if c.DbRole != "none" && c.DbRole != "discovery" {
+
+		if len(discoveryServerConfig.ClientConfig.DiscoveryServiceAddr) <= 0 ||
```

Comment:
DiscoveryServiceAddr is a string, instead of len check compare if its empty.

Reply:
You mean like discoveryServerConfig.ClientConfig.DiscoveryServiceAddr == "" ?
There is no inbuilt method in go which checks for empty string

Reply:
Yes